taskvine/src/manager/vine_file_replica_table.c

```diff
@@ -60,6 +60,45 @@ struct vine_worker_info *vine_file_replica_table_find_worker(struct vine_manager
 	return 0;
 }
 
+// return a set of up to q->temp_replica_count workers that do not have the file cachename
+// and are not on the same host as worker w.
+struct vine_worker_info **vine_file_replica_table_find_replication_targets(
+		struct vine_manager *q, struct vine_worker_info *w, const char *cachename, int *count)
+{
+	char *id;
+	struct vine_worker_info *peer;
+	struct vine_file_replica *remote_info;
+
+	int found = 0;
+	struct vine_worker_info **workers = malloc(sizeof(struct vine_worker_info) * (q->temp_replica_count));
+
+	HASH_TABLE_ITERATE(q->worker_table, id, peer)
```

Comment:
With this, workers listed at the start of the table will get all the replicas. Do you want to make this more uniform?

Reply:
Yes its not very sophisticated, it will only give the first worker q->worker_source_max_transfers before moving on to the next. If temp files are sporadically created it will only deliver to the first worker. I will think up some heuristic